docs/content/concepts/metadata-tags/asset-metadata/table-metadata.mdx

```diff
@@ -0,0 +1,192 @@
+2. Attach the `TableSchema` object to the asset as part of the `metadata` parameter under the `dagster/column_schema` key. This can be attached to your asset definition, or to the <PyObject object="MaterializeResult" /> object returned by the asset function
+
+Below are two examples of how to attach column schema metadata to an asset, one as definition metadata and one as materialization metadata:
+
+```python file=/concepts/metadata-tags/asset_column_schema.py
+from dagster import AssetKey, MaterializeResult, TableColumn, TableSchema, asset
+
+
+# Here, we know the schema of the asset, so we can attach it to the asset decorator
+@asset(
+    deps=[AssetKey("source_bar"), AssetKey("source_baz")],
+    metadata={
+        "dagster/column_schema": TableSchema(
+            columns=[
+                TableColumn(
+                    "name",
+                    "string",
+                    description="The name of the person",
+                ),
+                TableColumn(
+                    "age",
+                    "int",
+                    description="The age of the person",
+                ),
+            ]
+        )
+    },
+)
+def my_asset(): ...
+
+
+# Here, the schema isn't known until runtime
+@asset(deps=[AssetKey("source_bar"), AssetKey("source_baz")])
+def my_other_asset():
+    column_names = ...
+    column_types = ...
+
+    columns = [
+        TableColumn(name, column_type)
+        for name, column_type in zip(column_names, column_types)
+    ]
+
+    yield MaterializeResult(
+        metadata={"dagster/column_schema": TableSchema(columns=columns)}
+    )
+```
+
+The schema for `my_asset` will be visible in the Dagster UI. You may optionally attach <PyObject object="TableColumnConstraints"/> to each column to provide additional context about the values in the column:
+
+```python file=/concepts/metadata-tags/asset_column_schema_constraints.py
+from dagster import (
+    AssetKey,
+    MaterializeResult,
+    TableColumn,
+    TableColumnConstraints,
+    TableSchema,
+    asset,
+)
+
+
+@asset(
+    deps=[AssetKey("source_bar"), AssetKey("source_baz")],
+    metadata={
+        "dagster/column_schema": TableSchema(
+            columns=[
+                TableColumn(
+                    "name",
+                    "string",
+                    description="The name of the person",
+                ),
+                TableColumn(
+                    "age",
+                    "int",
+                    description="The age of the person",
+                    constraints=TableColumnConstraints(nullable=False, other=[">0"]),
+                ),
+            ]
+        )
+    },
+)
+def my_asset(): ...
+```
+
+### For assets loaded from integrations
+
+Column schemas are currently supported in the dbt integration. Refer to the [dbt documentation](/integrations/dbt/reference) for more information.
```

Comment:
I think this could be a little more specific: something like "Dagster's dbt integration supports automatically attaching column schema metadata to assets".

Also, is there a more direct subheading we could link to on that page?

Reply:
Isn't it this?

https://docs.dagster.io/integrations/dbt/reference#fetching-column-level-lineage

Reply:
Or https://docs.dagster.io/integrations/dbt/reference#customizing-metadata - this section mentions it as definition-level metadata (the above is materialization-time)